File: tests/example-frames/vertices-2.cpp
#include "ufbx.h"
#include <stdio.h>
#include <stdint.h>
#include <vector>

void output_triangle(ufbx_vec3 a, ufbx_vec3 b, ufbx_vec3 c)
{
    printf("(%.2f, %.2f, %.2f) (%.2f, %.2f, %.2f) (%.2f, %.2f, %.2f)\n",
        a.x, a.y, a.z, b.x, b.y, b.z, c.x, c.y, c.z);
}

// -- EXAMPLE_SOURCE --

int main(int argc, char **argv)
{
    ufbx_scene *scene = ufbx_load_file("my_scene.fbx", NULL, NULL);
    assert(scene);

    ufbx_node *node = ufbx_find_node(scene, "Cube");
    assert(node && node->mesh);

    list_triangles(node->mesh);

    return 0;
}

File: tests/example-frames/vertices-2.rs
use ufbx;
use std::vec::{Vec};

fn output_triangle(a: ufbx::Vec3, b: ufbx::Vec3, c: ufbx::Vec3) {
    println!("({:.2}, {:.2}, {:.2}) ({:.2}, {:.2}, {:.2}) ({:.2}, {:.2}, {:.2})",
        a.x, a.y, a.z, b.x, b.y, b.z, c.x, c.y, c.z);
}

// -- EXAMPLE_SOURCE --

fn main() {
    let scene = ufbx::load_file("my_scene.fbx", ufbx::LoadOpts::default()).unwrap();
    let node = scene.find_node("Cube").unwrap();
    list_triangles(node.mesh.as_ref().unwrap());
}
